rust/src/db.rs

```diff
@@ -196,3 +211,196 @@ pub async fn insert_ignore_txid(txid: Txid, maker_amount: i64) -> Result<()> {
 
     Ok(())
 }
+
+pub async fn insert_payment(payment: &PaymentInfo) -> Result<()> {
+    let mut conn = acquire().await?;
+
+    let PaymentInfo {
+        hash,
+        preimage,
+        secret,
+        status,
+        amt_msat,
+        created_timestamp,
+        updated_timestamp,
+    } = payment.clone();
+
+    let query_result = sqlx::query(
+        r#"
+        INSERT INTO payments (payment_hash, preimage, secret, htlc_status, amount_msat, created, updated)
+        VALUES ($1, $2, $3, $4, $5, $6, $7)
+        "#,
+    )
+    .bind(base64::encode(hash.0))
+    .bind(preimage.map(|x| base64::encode(x.0)))
+    .bind(secret.map(|x| base64::encode(x.0)))
+    .bind(status)
+        .bind(amt_msat.0.map(|msat| msat as i64))
+    .bind(created_timestamp as i64)
+    .bind(updated_timestamp as i64)
+    .execute(&mut conn)
+    .await?;
+
+    ensure!(
+        query_result.rows_affected() == 1,
+        "Failed to insert payment"
+    );
+    tracing::info!("Successfully stored new payment");
+    Ok(())
+}
+
+pub async fn update_payment(hash: PaymentHash, status: HTLCStatus) -> Result<()> {
+    let mut connection = acquire().await?;
+    let updated = time::OffsetDateTime::now_utc().unix_timestamp();
```

Comment:
🤔 Hmm, no I'm confused. I thought this timestamp was set on the outside?

Reply:
Yeah, we set this twice and effectively it will be what we set here. I think we have  to agree on what this timestamp represents. Is `updated` when we *received* the payment update, or is it (internally) use to mark when we last updated the row in the db?

Reply:
we're not using this new API anywhere yet! so far I just aligned the amount of fields, and the part of actual insertion / update will be changed. :)